objects/src/accounts/package/storage_entry/template.rs

```diff
@@ -0,0 +1,115 @@
+use alloc::{
+    string::{String, ToString},
+    vec::Vec,
+};
+
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use vm_core::{Felt, Word};
+use vm_processor::Digest;
+
+use crate::accounts::package::AccountComponentTemplateError;
+
+// TEMPLATE KEY
+// ================================================================================================
+
+// A simple wrapper type around a string key, used to enable templating.
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub struct TemplateKey {
+    key: String,
+}
+
+impl TemplateKey {
+    pub fn new(key: impl Into<String>) -> Self {
+        Self { key: key.into() }
+    }
+
+    pub fn inner(&self) -> &str {
+        &self.key
+    }
+}
+
+impl TryFrom<&str> for TemplateKey {
+    type Error = String;
+
+    fn try_from(value: &str) -> Result<Self, Self::Error> {
+        if value.starts_with("{{") && value.ends_with("}}") {
+            let inner = &value[2..value.len() - 2];
+            Ok(TemplateKey { key: inner.to_string() })
+        } else {
+            Err(format!("expected string in {{...}} format, got '{}'", value))
+        }
+    }
+}
+
+impl TryFrom<&String> for TemplateKey {
+    type Error = String;
+
+    fn try_from(value: &String) -> Result<Self, Self::Error> {
+        Self::try_from(value.as_str())
+    }
+}
+
+impl core::fmt::Display for TemplateKey {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        write!(f, "{}", self.key)
```

Comment:
I'm wondering if this should include braces. The inverse operation is basically `TryFrom<&str>` and it would expect braces, so maybe this would be most consistent.

Reply:
Yes, agreed, good catch. I originally did `From<&str>` instead of `TryFrom<&str>` which worked as the inverse of this, but then changed it based on Bobbin's suggestion and didn't realize it changed this.